adal/src/main/java/com/microsoft/aad/adal/AcquireTokenRequest.java

```diff
@@ -477,7 +478,15 @@ private AuthenticationResult acquireTokenSilentFlow(final AuthenticationRequest
             return authResult;
         }
 
-        //3. We couldn't get locally...If eligible return via broker... otherwise return local result
+        //3. try SAML Assertion
+        if (authenticationRequest.getSamlAssertion() != null) {
+            final AuthenticationResult authResultFromSaml = tryAcquireTokenSilentWithAssertion(authenticationRequest);
+            if (isAccessTokenReturned(authResultFromSaml)) {
+                return authResultFromSaml;
+            }            
+        }
```

Comment:
This behavior where we fall-thru and try the broker isn't 100% clear to me... want to confirm....

If the caller has supplied a SAML assertion in their request, and no access token was returned are we falling-thru in hopes that the broker has an RT/FRT/PRT for this user already?

Looking to confirm that the decision to proceed here and attempt the request in the broker in intentional

Reply:
> If the caller has supplied a SAML assertion in their request, and no access token was returned are we falling-thru in hopes that the broker has an RT/FRT/PRT for this user already?

I understand what you are saying. But there is a possibilty that there are no refresh tokens available in the cache, and the broker could provide them. In the former code flow, when the refresh token isnt found in the cache the application would try and use the broker to fetch tokens. So I am ensuring that this fall through behavior is present because those use-cases would not have any assertions. 

Now about trying saml assertion before the broker code flow, if the saml assertion is present than its very likely well be able to get the tokens using that, if its not present then the former fall through behavior would come into the picture. 
I maybe missing out something here, correct me if I am wrong

Reply:
Just wanted to clarify this was the desired behavior; sounds like 'yes'. Resolving this question, thanks